### PR TITLE
Introducing a new Context Spell Checker annotator

### PR DESCRIPTION
**Pull request by albertoandreottiATgmail**
## Description
+ Improved Python API.
+ Training available from Scala/Python.
+ Error dataset Generation.
+ New models.
+ New notebooks.
+ Error detection for performance.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code improvements with no or little impact
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](http://nlp.johnsnowlabs.com/contribute.html) page.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by maziyarpanahi** [comment]
@albertoandreottiATgmail The unit test failed here:
https://travis-ci.org/github/JohnSnowLabs/spark-nlp/jobs/682389107#L3395

There are also lots of training logs, the log size is limited in Travis so it would be good if to reduce them if not necessary.

File: src/main/scala/com/johnsnowlabs/nlp/annotators/spell/context/ContextSpellCheckerApproach.scala [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+      }
+    }
+
+    /* Blacklists {fwis, hyphen, slash} */
+    // words that appear with first letter capitalized (e.g., at the beginning of sentence)
+    val fwis = vocab.filter(_._1.length > 1).filter(_._1.head.isUpper).
+      filter(w => vocab.contains(w._1.head.toLower + w._1.tail)).keys
+
+    // Remove compound words for which components parts are in vocabulary, keep only the high frequent ones
+    val compoundWords = Seq("-", "/").flatMap { separator =>
```
Comment (jfernandrezj) [inline src/main/scala/com/johnsnowlabs/nlp/annotators/spell/context/ContextSpellCheckerApproach.scala:272]: can this list be a param?

Comment (maziyarpanahi) [reply src/main/scala/com/johnsnowlabs/nlp/annotators/spell/context/ContextSpellCheckerApproach.scala:272]: If this turns out to be a new param, let's do it in RC3 after I release RC2

**Comment by maziyarpanahi** [comment]
I rebased this with 250-release-candidate branch and resolved the conflicts.

**Comment by maziyarpanahi** [comment]
Tests have passed, I'll merge this and any additional changes + test of RC2 can happen after in RC3

**Pull request merged**